src/Window.cpp

```diff
@@ -1136,14 +1213,46 @@ QString Window::executeCommand(QString command) {
     else if (function == "wallLeft") {
         return boolToString(wallLeft());
     }
-    else if (function == "moveForward") {
+    else if (function == "moveFullForward") {
         int distance = 1;
         if (tokens.size() == 2) {
             distance = tokens.at(1).toInt();
         }
         bool success = moveForward(distance);
         return success ? "" : CRASH;
     }
+    else if (function == "moveHalfForward") {
+        int distance = 1;
+        if (tokens.size() == 2) {
+            distance = tokens.at(1).toInt();
+        }
+        bool success = moveHalfForward(distance);
+        return success ? "" : CRASH;
+    }
+    else if (function == "moveEdgeForward") {
+        int distance = 1;
+        if (tokens.size() == 2) {
+            distance = tokens.at(1).toInt();
+        }
+        bool success = moveEdgeForward(distance);
+        return success ? "" : CRASH;
+    }
+    else if (function == "completeRun") {
+        completeRun();
+        return ACK;
+    }
+    else if (function == "turnBack") {
+        turnBack();
+        return "";
+    }
+    else if (function == "turnRight45") {
+        turnRight45();
```

Comment:
What happens if there's a wall in the way? Could this movement result in a crash? Should we capture that in the API? Same goes for some other commands below.